File: isegoria/src/main/java/com/eulersbridge/isegoria/network/adapters/NullPrimitiveAdapter.java
package com.eulersbridge.isegoria.network.adapters;

import android.support.annotation.Nullable;

import com.squareup.moshi.FromJson;


@SuppressWarnings("unused")
public class NullPrimitiveAdapter {

    @FromJson
    public int intFromJson(@Nullable Integer value) {
        return value == null? 0 : value;
    }

    @FromJson
    public boolean booleanFromJson(@Nullable Boolean value) {
        return value == null? false : value;
    }

    @FromJson
    public double doubleFromJson(@Nullable Double value) {
        return value == null? 0 : value;
    }

    @FromJson
    public long longFromJson(@Nullable Long value) {
        return value == null? 0 : value;
    }
}

File: isegoria/src/main/kotlin/com/eulersbridge/isegoria/network/adapters/NullPrimitiveAdapter.kt
package com.eulersbridge.isegoria.network.adapters

import com.squareup.moshi.FromJson

class NullPrimitiveAdapter {

    @FromJson
    fun intFromJson(value: Int?) = value ?: 0

    @FromJson
    fun booleanFromJson(value: Boolean?) = value ?: false

    @FromJson
    fun doubleFromJson(value: Double?) = value ?: 0.0

    @FromJson
    fun longFromJson(value: Long?) = value ?: 0
}
